File: 202.py
def isHappy(n):
    seen = set()
    curr = str(n)
    while curr not in seen:
        seen.add(curr)
        sum = 0
        for elem in curr:
            elem = int(elem)
            sum +=elem**2;
        if sum==1:return True
        curr = str(sum)
    return False
            

File: 202.cpp
// is happy in leetcode , code solution below 
#include<bits/stdc++.h>
#include<iostream>

using namespace std;

bool isHappy(int n){
  set<int> s;
  string curr = to_string(n);
  while(!s.count(stoi(curr))){
    s.insert(stoi(curr));
    int sum =0;
    for(auto it: curr){
      int digit =it - '0';
      sum +=digit*digit;
    }
    if(sum==1){
      return true;
    }
    curr = to_string(sum);
  }
  return false;
}

int main{
  return 0;
}
